cmd/kubelet/app/options/options.go

```diff
@@ -181,6 +182,7 @@ func (s *KubeletServer) AddFlags(fs *pflag.FlagSet) {
 	fs.IntVar(&s.ImageGCHighThresholdPercent, "image-gc-high-threshold", s.ImageGCHighThresholdPercent, "The percent of disk usage after which image garbage collection is always run. Default: 90%")
 	fs.IntVar(&s.ImageGCLowThresholdPercent, "image-gc-low-threshold", s.ImageGCLowThresholdPercent, "The percent of disk usage before which image garbage collection is never run. Lowest disk usage to garbage collect to. Default: 80%")
 	fs.IntVar(&s.LowDiskSpaceThresholdMB, "low-diskspace-threshold-mb", s.LowDiskSpaceThresholdMB, "The absolute free disk space, in MB, to maintain. When disk space falls below this threshold, new pods would be rejected. Default: 256")
+	fs.DurationVar(&s.CalculateVolumeDiskUsageFrequency.Duration, "calculate-volume-disk-usage-interval", s.CalculateVolumeDiskUsageFrequency.Duration, "Specifies how often kubelet calculates and caches the volume disk usage for all pods and volumes.  Example: '5m'")
```

Comment:
SGTM.

On Thu, Feb 4, 2016 at 8:50 AM, Phillip Wittrock <notifications@github.com>
wrote:

> In cmd/kubelet/app/options/options.go
> <https://github.com/kubernetes/kubernetes/pull/19741#discussion_r51900721>
> :
>
> > @@ -181,6 +182,7 @@ func (s *KubeletServer) AddFlags(fs *pflag.FlagSet) {
> >  	fs.IntVar(&s.ImageGCHighThresholdPercent, "image-gc-high-threshold", s.ImageGCHighThresholdPercent, "The percent of disk usage after which image garbage collection is always run. Default: 90%")
> >  	fs.IntVar(&s.ImageGCLowThresholdPercent, "image-gc-low-threshold", s.ImageGCLowThresholdPercent, "The percent of disk usage before which image garbage collection is never run. Lowest disk usage to garbage collect to. Default: 80%")
> >  	fs.IntVar(&s.LowDiskSpaceThresholdMB, "low-diskspace-threshold-mb", s.LowDiskSpaceThresholdMB, "The absolute free disk space, in MB, to maintain. When disk space falls below this threshold, new pods would be rejected. Default: 256")
> > +	fs.DurationVar(&s.CalculateVolumeDiskUsageFrequency.Duration, "calculate-volume-disk-usage-interval", s.CalculateVolumeDiskUsageFrequency.Duration, "Specifies how often kubelet calculates and caches the volume disk usage for all pods and volumes.  Example: '5m'")
>
> If we encounter performance issues due to the run frequency, I'd like to
> be able to increase the frequency instead of forcing folks to turn it off
> completely. We can disable if the frequency is <= 0.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/19741/files#r51900721>.
>